### ISSUE TITLE
tensorflow for rocm

### ISSUE THREAD

**Issue by perestoronin** [justxi/rocm#175]
Please tell me where to download the ebuild to build tensorflow for rocm ?

PS. tensorflow requires that the rocm directory is not in /usr, but in /opt/rocm, but if targeted rocm directory at  /usr, then the tensorflow build script loops :(

**Comment by justxi** [comment]
I don´ t know. I don´t have any ebuilds for tensorflow.

**Comment by justxi** [comment]
Maybe @heroxbd @littlewu2508  have an ebuild for tensorflow?

Or did you mean pytorch?
But the ebuilds were not developed by me: https://github.com/justxi/rocm/tree/master/sci-libs/pytorch
-> https://github.com/justxi/rocm/issues/133

**Comment by perestoronin** [comment]
ebuild for tensorflow exists in portage - https://github.com/gentoo/gentoo/blob/master/sci-libs/tensorflow/tensorflow-2.4.0_rc1.ebuild but in ebuild masked rocm by flag internal https://github.com/gentoo/gentoo/blob/master/sci-libs/tensorflow/tensorflow-2.4.0_rc1.ebuild#217 export TF_NEED_ROCM=0

PS. also exists old intrestinginfrastucture of tensorflow for gentoo https://github.com/gridgentoo/gentoo-tensorflow-overlay/blob/master/ai-frameworks/tensorflow/tensorflow-1.4.0.ebuild

**Comment by perestoronin** [comment]
I see working in progress in portage tree - now added sys-devel/llvm-roc/files/llvm-roc-3.9.0-tensorflow.patch 
but sci-libs/tensorflow without rocm flag yet.

**Comment by perestoronin** [comment]
tensorflow new released 2.4.0 14-12-2020

**Comment by perestoronin** [comment]
rocm 4.0 released https://github.com/RadeonOpenCompute/ROCm/tags

**Comment by justxi** [comment]
> rocm 4.0 released https://github.com/RadeonOpenCompute/ROCm/tags

https://github.com/justxi/rocm/issues/177 ;-)

**Comment by perestoronin** [comment]
- tensorflow ebuild update to 2.5.0 need 
+ rocm ebuild updated to 4.2.0 done (to /opt/rocm-4.2.0 and symlink /opt/rocm), also added new ebuild sci-libs/hipFFT
+ cuda ebuild updated to 11.3.1 done

and when try to build tensorflow with flag rocm=1 : gcc: fatal error: cannot execute ‘cc1plus’: execvp: No such file or directory

how to fix with trouble about cc1plus ?
